R/zchunk_L242.Bld_Inc_Elas_scenarios.R

```diff
@@ -34,55 +44,185 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios_DISABLED <- function(command,
     A42.demand <- get_data(all_data, "energy/A42.demand")
     A42.inc_elas <- get_data(all_data, "socioeconomics/A42.inc_elas")
     L101.Pop_thous_GCAM3_R_Y <- get_data(all_data, "L101.Pop_thous_GCAM3_R_Y")
-    L102.gdp_mil90usd_GCAM3_R_Y <- get_data(all_data, "L102.gdp_mil90usd_GCAM3_R_Y")
-    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+    L102.gdp_mil90usd_GCAM3_R_Y <- get_data(all_data, "temp-data-inject/L102.gdp_mil90usd_GCAM3_R_Y") %>%
+      # Temporary for temp-data-inject
+      gather(year, gdp, starts_with('X')) %>%
+      mutate(year = as.integer(substr(year,2,5)))
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "temp-data-inject/L102.pcgdp_thous90USD_Scen_R_Y") %>%
+      # Temporary for temp-data-inject
+      gather(year, pcgdp_90thousUSD, starts_with('X')) %>%
+      mutate(year = as.integer(substr(year,2,5)))
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Joining population and GDP data in order to calculate per capita GDP
+    L242.pcgdp_GCAM3_R_Y <- L101.Pop_thous_GCAM3_R_Y %>%
+      rename(population = value) %>%
+      left_join_error_no_match(L102.gdp_mil90usd_GCAM3_R_Y, by = c("GCAM_region_ID","year")) %>%
+      mutate(pcgdp_90thousUSD = gdp/population) %>%
+      select(-population, -gdp) %>%
+      filter(year %in% FUTURE_YEARS)
+
+    # Linearly interpolate income elasticity for each level of per-capita GDP,
+    # using the assumption data
+    L242.IncomeElasticity_bld_GCAM3 <- L242.pcgdp_GCAM3_R_Y %>%
+      mutate(income.elasticity = approx(x = A42.inc_elas$pcgdp_90thousUSD, y = A42.inc_elas$inc_elas,
```

Comment:
I may be wrong in my understanding of approx_fun, but I don't think that would work here. This is taking the gdp and income elasticity from another file and using that relationship to estimate income elasticity for different gdp data. Basically creating a linear model to generate predictions. That doesn't look to me what approx_fun is capable of handling.